java/core/src/main/java/sleeper/core/properties/model/DataEngine.java

```diff
@@ -24,7 +24,12 @@
 public enum DataEngine {
     /** Pure Java implementations. */
     JAVA,
-    /** Uses a native library written in Rust to perform data operations with Apache DataFusion. */
+    /** Uses a native library written in Rust to perform compactions with Apache DataFusion. */
+    DATAFUSION_COMPACTION_ONLY,
+    /**
+     * Uses a native library written in Rust to perform data operations with Apache DataFusion.
+     * Performing queries with DataFusion is experimental.
+     */
     DATAFUSION;
```

Comment:
DATAFUSION is the default value, which means instances will enable DataFusion for queries by default. Maybe we should have something like a DATAFUSION_EXPERIMENTAL value?